test/Base_Tests/src/Semantic/Any_Spec.enso

```diff
@@ -38,75 +38,89 @@ add_specs suite_builder =
             With_Y.x . to_text . should_equal "Any:With_Y"
-            With_Y.y.should_be_a Function
+            Test.expect_panic_with (With_Y.y) No_Such_Method
             With_X.to_text . to_text . should_equal "With_X"
             With_Y.to_text . to_text . should_equal "With_Y"
             Any.x . to_text . should_equal "Any:Any"
+
+        group_builder.specify "Static method invocation (x)" <|
             Any.x self=Any . to_text . should_equal "Any:Any"
             Any.x self=With_X . should_equal "Any:With_X"
             Any.x self=With_Y . should_equal "Any:With_Y"
-            Any.x (My_Type.Value 22) . should_equal "Any:(My_Type.Value 22)"
-            Any.x (With_X.Value 22) . should_equal "Any:(With_X.Value 22)"
-            Any.x (With_Y.Value 22) . should_equal "Any:With_Y(22)"
+            Test.expect_panic_with (Any.x (My_Type.Value 22)) Not_Invokable
+            Test.expect_panic_with (Any.x (With_X.Value 22)) Not_Invokable
+            Test.expect_panic_with (Any.x (With_Y.Value 22)) Not_Invokable
             Date.to_display_text . should_equal "Date"
+            Any.x self=(My_Type.Value 22) . should_equal "Any:(My_Type.Value 22)"
+            Any.x self=(With_X.Value 22) . should_equal "Any:(With_X.Value 22)"
+            Any.x self=(With_Y.Value 22) . should_equal "Any:With_Y(22)"
 
-        group_builder.specify "static method calls on Any should have defaulted self argument to Any" <|
+        group_builder.specify "Static method invocation (==)" <|
             # Any.== is a method that takes two arguments (self other)
             (Any.==) . should_be_a Function
             (Any.== self=Boolean) . should_be_a Function
             (Any.== self=Any Boolean) . should_be_false
-            (Any.== Any Any) . should_be_true
+            Test.expect_panic_with (Any.== Any Any) Not_Invokable
             (Any.== Boolean) . should_be_false
             (Any.== Any) . should_be_true
-            (Any == Boolean) . should_be_false
+            (Any.== self=Any Any) . should_be_true
+
+        group_builder.specify "Instance method invocation (==)" <|
             (Any == Any) . should_be_true
+            (Any == Boolean) . should_be_false
 
+        group_builder.specify "Static method invocation (to_text)" <|
             # Any.to_text is a method that takes one argument (self)
             Any.to_text . should_equal "Any"
+            Test.expect_panic_with (Any.to_text Any) Not_Invokable
+            Test.expect_panic_with (Any.to_text Vector) Not_Invokable
             Any.to_text self=Any . should_equal "Any"
             Any.to_text self=Vector . should_equal "Vector"
             Any.to_text self=[1,2,3] . should_equal "[1, 2, 3]"
-            Any.to_text Any . should_equal "Any"
-            Any.to_text Vector . should_equal "Vector"
-            Any.to_text [1,2,3] . should_equal "[1, 2, 3]"
+            Test.expect_panic_with (Any.to_text Any) Not_Invokable
```

Comment:
Not_Invokable because `Any.to_text Any` is equivalent to `Any.to_text self=Any Any` and thus evaluates to `"Any" Any`